src/scheduler/ExecGraph.cpp

```diff
@@ -70,6 +71,45 @@ std::vector<std::string> getMpiRankHostsFromExecGraph(const ExecGraph& graph)
     return getMpiRankHostsFromExecGraphNode(graph.rootNode);
 }
 
+void getMigratedMpiRankHostsFromExecGraph(const ExecGraph& graph,
+                                          std::vector<std::string>& hostsBefore,
+                                          std::vector<std::string>& hostsAfter)
+{
+    std::queue<faabric::scheduler::ExecGraphNode> nodeList;
+    nodeList.push(graph.rootNode);
+    while (!nodeList.empty()) {
+        // Process the node at the front
+        auto node = nodeList.front();
+        int returnValue = node.msg.returnvalue();
+        int rank = node.msg.mpirank();
+        std::string executedHost = node.msg.executedhost();
+        if (returnValue == 0) {
```

Comment:
There's 26 lines and up to 3 levels of nesting in this code block, but no linebreaks. A bit of whitespace would make it easier to parse.